python_modules/libraries/dagster-fivetran/dagster_fivetran/assets.py

```diff
@@ -0,0 +1,99 @@
+    :py:class:`~dagster_fivetran.fivetran_resource`, which allows it to communicate with the
+    Fivetran API.
+
+    Args:
+        connector_id (str): The Fivetran Connector ID that this op will sync. You can retrieve this
+            value from the "Setup" tab of a given connector in the Fivetran UI.
+        asset_keys (str): The set of asset keys that Dagster will expect this asset to produce.
+            These should be of the form ``AssetKey([<schema_name>, <table_name>])`` where
+            ``<schema_name>`` and ``<table_name>`` represent the location of the tables in the
+            Fivetran destination. Any AssetKey listed here MUST be sync'd every time this computation
+            is executed, otherwise an exception will be raised. Additional tables that are updated
+            during the sync but not listed here will be recorded with runtime AssetMaterialization
+            events.
+        name (Optional[str]): A name for the underlying computation.
+        poll_interval (float): The time (in seconds) that will be waited between successive polls.
+        poll_timeout (Optional[float]): The maximum time that will waited before this operation is
+            timed out. By default, this will never time out.
+        io_manager_key (Optional[str]): The io_manager to be used to handle each of these assets.
+
+    Examples:
+
+    .. code-block:: python
+
+        from dagster import AssetKey
+        from dagster.core.asset_defs import build_assets_job
+
+        from dagster_fivetran import fivetran_resource
+        from dagster_fivetran.assets import fivetran_assets_factory
+
+        my_fivetran_resource = fivetran_resource.configured(
+            {
+                "api_key": {"env": "FIVETRAN_API_KEY"},
+                "api_secret": {"env": "FIVETRAN_API_SECRET"},
+            }
+        )
+
+        fivetran_assets = fivetran_assets_factory(connector_id="foobar", asset_keys=[
+            AssetKey(["schema1", "table1"]), AssetKey(["schema2", "table2"])
+        ])
+
+        my_fivetran_job = build_assets_job(
+            "my_fivetran_job",
+            assets=[fivetran_assets],
+            resource_defs={"fivetran": my_fivetran_resource}
+        )
+
+    """
+
+    @multi_asset(
+        name=name or f"fivetran_sync_{connector_id}",
+        outs={
+            key.path[-1]: Out(io_manager_key=io_manager_key, asset_key=key) for key in asset_keys
+        },
+        required_resource_keys={"fivetran"},
+    )
+    def _assets(context):
+        fivetran_output = context.resources.fivetran.sync_and_poll(
+            connector_id=connector_id,
+            poll_interval=poll_interval,
+            poll_timeout=poll_timeout,
+        )
+        for materialization in generate_materializations(fivetran_output, asset_key_prefix=[]):
+            # scan through all tables actually created, if it was expected then emit an Output.
```

Comment:
I think representing this description with test assertions would help make me more confident about the behavior of passing `asset_keys`. Specifically, it sounds like we should be asserting against outputs.